### PR TITLE
Update Discovery docs with a realistic example.

### LINKED ISSUE #1135: Improve discovery docs example
The example in the discovery feature documentation should show a realistic configuration. The current example shows how to enable decision logging but nothing else. This isn't very practical. Let's replace it with an example that shows how an OPA can be dynamically configured to download one of two bundles based on an environment variable that the OPA is started with.

### PR DESCRIPTION
**Pull request by ashutosh-narkar**
Fixes #1135 

Signed-off-by: Ashutosh Narkar <anarkar4387@gmail.com>

### CODE REVIEW

File: docs/book/discovery.md [hunk 1]
```diff
@@ -103,3 +103,101 @@ In both cases, OPA's configuration is hierarchically organized under the
...
+{%ace lang='python'%}
+package example
+
+discovery = {
+  "bundle": {
+    "name": bundle_name
+  }
+}
+
+rt = opa.runtime()
```
Comment (tsandall) [inline docs/book/discovery.md]: Nit: I'd probably just make these into the body of the discovery rule. Either works.

File: docs/book/discovery.md [hunk 2]
```diff
@@ -103,3 +103,101 @@ In both cases, OPA's configuration is hierarchically organized under the
...
+region = rt["config"]["labels"]["region"]
```
Comment (tsandall) [inline docs/book/discovery.md]: Nit: `rt.config.labels.region`

File: docs/book/discovery.md [hunk 3]
```diff
@@ -103,3 +103,101 @@ In both cases, OPA's configuration is hierarchically organized under the
 `discovery.name` value. If discovery is enabled, the `service` field in the
 `bundles`, `status`, `decision logs` plugins is optional and will default to one
 of the services from the discovery configuration.
+
+## Example
+
+Let's see an example of how the discovery feature can be used to dynamically configure an OPA to download one of two bundles based on an environment variable that the OPA was started with. Let's say the environment variable `region` indicates the region in which the OPA is running and it's value will decide the bundle to download.
```
Comment (tsandall) [inline docs/book/discovery.md]: Nit: This example is using configuration labels instead of environment variables.

File: docs/book/discovery.md [hunk 4]
```diff
@@ -103,3 +103,101 @@ In both cases, OPA's configuration is hierarchically organized under the
...
+  }
+}
+
+rt = opa.runtime()
+region = rt["config"]["labels"]["region"]
+bundle_name = region_bundle[region]
+
+# region-bundle information
+region_bundle = {
+  "US": "disco/test1/p",
```
Comment (tsandall) [inline docs/book/discovery.md]: Nit: change `disco` to `example` or something other than `disco` to indicate that this is the -actual- bundle that OPA is going to activate and use to serve queries.

File: docs/book/discovery.md [hunk 5]
```diff
@@ -103,3 +103,101 @@ In both cases, OPA's configuration is hierarchically organized under the
...
+
+```raw
+INFO Bundle downloaded and activated successfully. name=disco/test2/p plugin=bundle
+```
+
+This shows how the discovery feature can help in centrally managing the bundle to be downloaded by an OPA based on an environment variable. You can use the same strategy to dynamically configure other plugins based on the running OPA's environment variables.
+
+## Limitations
+
+The discovery feature cannot be used to dynamically modify `services`, `labels`, `default_decision` , `default_authorization_decision` and `discovery`. This means that these configuration settings should be included in the bootup configuration file provided to OPA.
```
Comment (tsandall) [inline docs/book/discovery.md]: Nit: default_decision and default_authorization_decision are supported as of https://github.com/open-policy-agent/opa/commit/2d425494aa3d4149c009d91febb5bfcff56f2734

Comment (ashutosh-narkar) [reply docs/book/discovery.md]: Updated comment.

File: docs/book/discovery.md (hunk 4 above)
Comment (ashutosh-narkar) [reply docs/book/discovery.md]: Done.

File: docs/book/discovery.md (hunk 3 above)
Comment (ashutosh-narkar) [reply docs/book/discovery.md]: Updated comment.

File: docs/book/discovery.md (hunk 2 above)
Comment (ashutosh-narkar) [reply docs/book/discovery.md]: Done.

**Pull request merged**